Message action sample widget (#59)
Esri/arcgis-experience-builder-sdk-resources: samples/widgets/subscribe/README.md

# Subscribe widget

This sample demonstrates how to use message action in widget.

## How to use the sample
Clone the [sample repo](https://github.com/esri/arcgis-experience-builder-sdk-resources) and copy this widget's folder (within `samples/widgets`) to the `client/your-extensions/widgets` folder of your Experience Builder installation.

## How it works

>   Within `query-action-setting.tsx`,create an class extends the React.PureComponent class with the types `ActionSettingProps`,

Reviewer: ```suggestion
  Within `query-action-setting.tsx`, create a class extends the React.PureComponent class with the type `ActionSettingProps`.
```

>   Within `query-action-setting.tsx`,create an class extends the React.PureComponent class with the types `ActionSettingProps`,

Reviewer 2: ```suggestion
Within `query-action-setting.tsx`,create a class extends the React.PureComponent class with the types `ActionSettingProps`,
```

>   the configuration of the message action can be released through the `onSettingChange` method

Reviewer: ```suggestion
  The configuration of the message action can be released through the `onSettingChange` method.
```

>   the configuration of the message action can be released through the `onSettingChange` method

Reviewer 2: ```suggestion
the configuration of the message action can be changed through the `onSettingChange` method
```

  ```javascript
this.props.onSettingChange({
  actionId: this.props.actionId,
  config: this.props.config.set('fieldName', field['name']).set('useDataSource',{
    dataSourceId: this.props.config.useDataSource.dataSourceId,
    mainDataSourceId: this.props.config.useDataSource.mainDataSourceId,
    dataViewId: this.props.config.useDataSource.dataViewId,
    rootDataSourceId: this.props.config.useDataSource.rootDataSourceId,
    fields: allSelectedFields.map(f => f.jimuName)
  })
});

```

> Within `query-action.tsx`, create an class extends the `AbstractMessageAction` class, we should set the uri of the action setting component through the `getSettingComponentUri` method, the `filterMessageType` method can filter out actions that do not meet our requirements. We can monitor the trigger of

Reviewer: ```suggestion
Within `query-action.tsx`, create a class extends the `AbstractMessageAction` class. Then set the URI of the action setting component through the `getSettingComponentUri` method. Use the `filterMessageType` method to filter out actions that are not relevant to this widget. The `onExecute` method is called when the action is triggered.
```

> the action in the `onExecute` method, and can perform the operation what we want

Reviewer: ```suggestion
```
